actionview/lib/action_view/helpers/form_with_helper.rb

```diff
@@ -0,0 +1,196 @@
+        end
+
+        def text_area(method, *args, **options)
+          options = options_for(method, options)
+          content = model.send(method) if model
+          content = args[0] if args.size > 0
+          content.nil? ? tag.textarea(options) : tag.textarea(content, options)
+        end
+
+        def label(method, *args, **options, &block)
+          content ||= args[0] if args.size > 0
+          content ||= I18n.t("#{model.model_name.i18n_key}.#{method}", default: "", scope: "helpers.label").presence if model && model.model_name
+          content ||= translate_with_human_attribute_name(method)
+          content ||= method.to_s.humanize
+          content = @template.capture(content, &block) if block_given?
+          tag.label content, options
+        end
+
+        def check_box(method, include_hidden: true, on: "1", off: "0", **options)
+          hidden = "".html_safe
+          options = options_for(method, options)
+          checkbox_options = options.merge(type: 'checkbox', value: on)
+          checkbox_options.merge!(checked: "checked") if model && checked?(on, model.send(method))
+          include_hidden = false if off.nil?
+          if include_hidden
+            hidden_options = options.merge(type: 'hidden', value: off)
+            hidden_options.delete(:checked)
+            hidden = tag.input(hidden_options).html_safe
+          end
+          hidden + tag.input(checkbox_options).html_safe
+        end
+
+        def select(method, choices = nil, blank: nil, prompt: nil, index: :undefined, disabled: nil, **options, &block)
+          tag.select option_tags_for_select(choices, blank: blank), options_for(method, options)
+        end
+
+        def collection_select(method, collection, value_method, text_method, blank: nil, prompt: nil, index: :undefined, disabled: nil, **options)
+          choices = collection.map do |object|
+            [object.send(value_method), object.send(text_method)]
+          end
+          select(method, choices, options)
+        end
+
+        def option_tags_for_select(choices, blank: false)
+          result = (blank ? tag.option("", value: "") : "").html_safe
+          result += choices.map do |choice|
+            if choice.is_a? Array
+              tag.option choice[0], value: choice[1]
+            else
+              tag.option choice, value: choice
+            end
+          end.join("\n").html_safe
+          result
+        end
+
+        private
+          def checked?(on_value, value)
+            case value
+            when TrueClass, FalseClass
+              value == !!on_value
+            when NilClass
+              false
+            when String
+              value == on_value
+            else
+              if value.respond_to?(:include?)
+                value.include?(on_value)
+              else
+                value.to_i == on_value.to_i
+              end
+            end
```

Comment:
Each case is covered with test corresponding test. However I am not sure if we need all the different cases.

I think it would be easier and more efficient if you start code review from tests, so that we discuss interface and different variants & cases -> then I implement conclusions -> and the we go back to implementation details do the review and implement changes.

There is still plenty open questions regarding usage that focusing on implementation might be to early.

What do you think? :)